jenkinsfiles/auth0-users/role.py

```diff
@@ -0,0 +1,65 @@
+import requests
+import logging
+
+
+logging.basicConfig()
+LOG = logging.getLogger(__name__)
+LOG.setLevel(logging.DEBUG)
+
+
+class Role(object):
+
+    def __init__(self, authz_api, authz_token, role_data):
+        if not role_data:
+            raise ValueError("role_data can't be empty")
+
+        self.authz_api = authz_api
+        self.authz_token = authz_token
+        self.data = role_data
+
+    def id(self):
+        return self.data["_id"]
+
+    def name(self):
+        return self.data["name"]
+
+    def description(self):
+        return self.data["description"]
+
+    def application_id(self):
+        return self.data["applicationId"]
+
+    def application_type(self):
+        return self.data["applicationType"]
+
+    def permission_ids(self):
+        return self.data["permissions"]
+
+
+    def add_permission(self, permission_id):
+        payload = {
+            'name': self.name(),
+            'description': self.description(),
+            'applicationId': self.application_id(),
+            'applicationType': self.application_type(),
+            'permissions': self.permission_ids(),
+        }
```

Comment:
I just realised why I can't, it's because the API will complain about other attributes.